scipy/special/specfun.c

```diff
@@ -0,0 +1,8417 @@
+            bl[k] = f;
+        }
+        if (k % 2 == 0) {
+            bs += 2.0 * f;
+        }
+        f0 = f1;
+        f1 = f;
+    }
+    bg = bs - f;
+    for (k = 0; k <= *nm; k++) {
+        bl[k] /= bg;
+    }
+    r0 = 1.0;
+    for (k = 1; k <= *nm; k++) {
+        r0 = 2.0 * r0 * k / x;
+        bl[k] *= r0;
+    }
+    dl[0] = -0.5 * x * bl[1];
+    for (k = 1; k <= *nm; k++) {
+        dl[k] = 2.0 * k / x * (bl[k - 1] - bl[k]);
+    }
+    return;
+}
+
+
+void specfun_lamv(double v, double x, double *vm, double *vl, double *dl) {
+
+    // =========================================================
+    // Purpose: Compute lambda function with arbitrary order v,
+    //          and their derivative
+    // Input :  x --- Argument of lambda function
+    //          v --- Order of lambda function
+    // Output:  VL(n) --- Lambda function of order n+v0
+    //          DL(n) --- Derivative of lambda function
+    //          VM --- Highest order computed
+    // Routines called:
+    //      (1) MSTA1 and MSTA2 for computing the starting
+    //          point for backward recurrence
+    //      (2) GAM0 for computing gamma function (|x| ≤ 1)
+    // =========================================================
+
+    int i, n, k, j, k0, m;
+    double cs, ga, fac, r0, f0, f1, f2, f, xk, vv;
+    double x2, v0, vk, bk, r, uk, qx, px, rp, a0, ck, sk, bjv0, bjv1;
+    const double pi = 3.141592653589793;
+    const double rp2 = 0.63661977236758;
+
+    x = fabs(x);
+    x2 = x * x;
+    n = (int)v;
+    v0 = v - n;
+    *vm = v;
+
+    if (x <= 12.0) {
+        for (k = 0; k <= n; k++) {
+            vk = v0 + k;
+            bk = 1.0;
+            r = 1.0;
+
+            for (i = 1; i <= 50; i++) {
+                r = -0.25 * r * x2 / (i * (i + vk));
+                bk = bk + r;
+
+                if (fabs(r) < fabs(bk) * 1.0e-15)
+                    break;
+            }
+            vl[k] = bk;
+
+            uk = 1.0;
+            r = 1.0;
+
+            for (i = 1; i <= 50; i++) {
+                r = -0.25 * r * x2 / (i * (i + vk + 1.0));
+                uk = uk + r;
+
+                if (fabs(r) < fabs(uk) * 1.0e-15)
+                    break;
+            }
+            dl[k] = -0.5 * x / (vk + 1.0) * uk;
+        }
+        return;
+    }
+
+    k0 = (x >= 35.0) ? 10 : ((x >= 50.0) ? 8 : 11);
+    bjv0 = 0.0;
+    bjv1 = 0.0;
+
+    for (j = 0; j <= 1; j++) {
+        vv = 4.0 * (j + v0) * (j + v0);
+        px = 1.0;
+        rp = 1.0;
+        for (k = 1; k <= k0; k++) {
+            rp = -0.78125e-2 * (vv - pow((4.0 * k - 3.0), 2.0)) * (vv - pow((4.0 * k - 1.0), 2.0)) / (k * (2.0 * k - 1.0) * x2);
+            px += rp;
+        }
+
+        qx = 1.0;
+        rp = 1.0;
+        for (k = 1; k <= k0; k++) {
+            rp = -0.78125e-2 * rp * (vv - pow((4.0 * k - 1.0), 2.0)) * (vv - pow((4.0 * k + 1.0), 2.0)) / (k * (2.0 * k + 1.0) * x2);
+            qx += rp;
+        }
+
+        qx = 0.125 * qx * (vv - 1.0) / x;
```

Comment:
```suggestion
        qx = 0.125 * (vv - 1.0) * qx / x;
```

https://github.com/scipy/scipy/blob/829d0dd950d63caece04526b3a617bbe2bc24023/scipy/special/specfun/specfun.f#L2713